### ISSUE TITLE
accessibility options and newer themes

### ISSUE THREAD

**Issue by ronm123** [thexerteproject/xerteonlinetoolkits#779]
@torinfo @FayCross 
I've committed the CESI theme now and following our discussion re the accessibility options and js code for that theme that isn't working at the moment I did some testing with the other newer themes that we have and there are definitely some issues with some of those too. The main issue is when selecting Invert. e.g. we have so many more colours and changes in themes now that the code behind that option isn't changing everything that is necessary to change and I'm not sure how easy it would be to do so. It's not just the newer themes there are some issues with some of the older themes too.

I'm wondering about the following and perhaps for further discussion and perhaps event development during the developer day next week...

Add new themes e.g. High contrast, Black on Yellow and have the accessibility options switch to those themes rather than try to invert colours in the existing theme.
and/or
Provide a select list of all the enabled themes so that a user can select from any of those themes if enabled by the author e.g. same option as disable theme change on url.
Possibly add a footer button for easily changing theme
Possibly include a browser cookie when viewing LO's to save choice of preferred theme to apply across all XOT projects

As I say maybe discuss further during the developer day.

**Comment by ronm123** [comment]
Hi all
I committed a new High Contrast theme yesterday and added the following notes:
New High Contrast XOT template theme with the aim of replacing the Colour Changer > Invert option or rather what happens when that option is selected. That option hasn't really worked reliably for some time because we have so many more themes and variations compared with just the original default theme. I guess this needs agreement/discussion as to whether to change the invert option to switch to this theme when selected. On the one hand it would provide a more reliable high contrast solution but on the other hand does change the theme (including buttons etc) rather than just invert white and black.

Do we agree this is a better option?
If so I'll also work on a Black on Yellow theme too for that colour changer option.
Obviously changes need to be made to the colour changer code so that the theme is switched rather than the current invert functionality.

**Comment by ronm123** [comment]
Hi all
I've committed the blackonyellow theme now too.
If we all agree then the colour changer code now needs changing so that if invert is selected the high contrast theme is loaded and if black on yellow is selected then the black on yellow theme is loaded.

**Comment by ronm123** [comment]
@torinfo thanks for updating the accessibility options for black on yellow and invert to swich to the new themes. That bit works fine but a slight problem that we have missed is that after choosing one of those options and then selecting off the theme doesn't revert to or reload the orginal theme set by the author. Refreshing obviously achieves that but ideally we need to add code so that isn't needed.

**Comment by ronm123** [comment]
BTW selecting to remove background images and then selecting off doesn't work now either.

**Comment by torinfo** [comment]
@ronm123 Hmmm. The code is not in there. I'll find it and will add it.

**Comment by ronm123** [comment]
@torinfo great thanks I see that you have now committed the change and after a clearing of cache it all seems to be working now so will close the issue.
